serve/mlc_serve/engine/engine_common.py

```diff
@@ -206,10 +169,54 @@ def check_stopping_sequences(stopping_criteria, output_text, delta, is_ended):
     return output_text, delta, is_ended
 
 
-def update_sequence(
+def prepare_logprob(
+    logprob_info,
+    delta,
+    gen_seq: GenerationSequence,
+    prompt_token_ids,
+    tokenizer,
+):
+    if logprob_info is None:
+        return []
+
+    outputs = []
+    for info in logprob_info:
+        assert info is not None
+        assert info.top_token_ids is not None
+        assert info.top_logprobs is not None
```

Comment:
parameter `top_logprobs` can be 0. It means a client needs only logprob for generated token. Not assert simply skip calculation if `info.top_logprobs is None`